improver_tests/calibration/dataframe_utilities/test_dataframe_utilities.py

```diff
@@ -397,6 +403,116 @@ def test_basic(self):
+                self.training_length,
+            )
+
+    def test_error_multiple_columns(self):
+        """Test that an error is raised if dataframe contains more than one of
+        REPRESENTATION_COLUMNS."""
+        msg = "More than one column"
+        df = self.forecast_df.copy()
+        df["realization"] = 0
+        with self.assertRaisesRegex(ValueError, msg):
+            forecast_and_truth_dataframes_to_cubes(
+                df,
+                self.truth_subset_df,
+                self.cycletime,
+                self.forecast_period,
+                self.training_length,
+            )
+
+    def test_station_id(self):
+        """Test that when station_id is present, both station_id and wmo_id
+        are used to match sites between forecast and truth data."""
+        forecast_df = self.forecast_df.copy()
+        forecast_df = forecast_df.loc[forecast_df["wmo_id"].isin(self.wmo_ids[:-1])]
+        forecast_df["station_id"] = forecast_df["wmo_id"] + "0"
+        truth_df = self.truth_subset_df.copy()
+        truth_df = truth_df.loc[truth_df["wmo_id"].isin(self.wmo_ids[1:])]
+        truth_df["station_id"] = truth_df["wmo_id"] + "0"
+        site_id_values = np.array([0], dtype=np.int32)
+        expected_forecast = self.expected_period_forecast[:, :, [1]].copy()
+        expected_forecast.coord("spot_index").points = site_id_values
+        unique_id_coord = iris.coords.AuxCoord(
+            [self.wmo_ids[1] + "0"],
+            long_name="station_id",
+            units="no_unit",
+            attributes={"unique_site_identifier": "true"},
+        )
+        site_id_dim = expected_forecast.coord_dims("spot_index")[0]
+        expected_forecast.add_aux_coord(unique_id_coord, site_id_dim)
+        expected_truth = self.expected_period_truth[:, [1]].copy()
+        site_id_dim = expected_truth.coord_dims("spot_index")[0]
+        expected_truth.add_aux_coord(unique_id_coord, site_id_dim)
+        expected_truth.coord("spot_index").points = site_id_values
+        result = forecast_and_truth_dataframes_to_cubes(
+            forecast_df,
+            truth_df,
+            self.cycletime,
+            self.forecast_period,
+            self.training_length,
+        )
+        self.assertEqual(len(result), 2)
+        self.assertCubeEqual(result[0], expected_forecast)
+        self.assertCubeEqual(result[1], expected_truth)
+        forecast_df["wmo_id"] = "00000"
+        truth_df["wmo_id"] = "00000"
+        expected_truth.coord("wmo_id").points = ["00000"]
+        expected_forecast.coord("wmo_id").points = ["00000"]
+        result = forecast_and_truth_dataframes_to_cubes(
+            forecast_df,
+            truth_df,
+            self.cycletime,
+            self.forecast_period,
+            self.training_length,
+        )
+        self.assertEqual(len(result), 2)
+        self.assertCubeEqual(result[0], expected_forecast)
+        self.assertCubeEqual(result[1], expected_truth)
```

Comment:
One query might be whether this could be moved into a separate unit test? This could be achieved by moving the shared code used to set up the required `forecast_df`, `truth_df`, `expected_forecast` and `expected_truth`. 

If that approach isn't preferred, it would be good to extend the docstring and/or add some comments to describe the two separate tests within this unit test i.e. the first test is using "valid" WMO IDs, whilst the second test is using a placeholder value for the WMO_ID. For the first test, the WMO_ID is primarily used for the matching whilst in the second test, as all sites have been given the same WMO ID, the station_id is vital for the site matching. It would be worth making it clear what is being tested here.

Reply:
I've refactored this into 2 tests.